### PR TITLE
Add form to quickly generate new crossword puzzles and add them to the smart contract

### PR DESCRIPTION
**Pull request by miancadonc**
Added form component that can generate a sample crossword that matches what would be pushed to smart contract. User can then modify prize or crossword before pushing. Made minor changes to crossword page and success page, plus changed rendering in App.js to accommodate form. Added to utils.js and created add_puzzle.js and signin.js to help with form functionality.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -23,6 +25,58 @@ const App = ({ nearConfig, data }) => {
...
+      if (apiInstance.isSignedIn() && window.location.pathname === "/success") {
+        setShowForm(true)
+        // nearAPI modifies history asynchronously as part of WalletConnection constructor
+        // https://github.com/near/near-api-js/issues/817
+        window.setTimeout(() => history.replaceState(null, '', '/'), 500)
+      }
+    })
+  }, [])
+
+  const renderPage = () => {
```
Comment (samingle) [inline src/App.js:41]: It'd be nice to wrap this in a useMemo so it's not doing unnecessary render passes

File: src/add-puzzle.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export const addNewPuzzle = async (mungedLayout, layout, dimensions, prizeDeposit) => {
+    const nearConfig = getConfig(process.env.NEAR_ENV || 'testnet');
+    const keyStore = new keyStores.BrowserLocalStorageKeyStore();
+    nearConfig.keyStore = keyStore;
+    const near = await connect(nearConfig);
+    const wallet = new WalletConnection(near);
+
+    const cleanLayout = layout.map(clueAnswer => {
+        // remove answer and capitalize direction value
```
Comment (samingle) [inline src/add-puzzle.js]: would be nice to explain why we're capitalizing things here in this comment so future folks don't have to dig through the rust code to understand it

File: src/signin.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import getConfig from "./config";
+import { connect, keyStores, WalletConnection } from "near-api-js";
+
+class ApiManager {
```
Comment (samingle) [inline src/signin.js]: we should make the file name match the export here, i'd think

Comment (miancadonc) [reply src/signin.js]: done

File: src/add-puzzle.js (hunk 2 above)
Comment (miancadonc) [reply src/add-puzzle.js]: done

**Pull request merged**